### PR TITLE
GmosNorthLongSlit sequence generation

### PR DESCRIPTION
**Pull request by swalker2m**
This PR implements a first pass at sequence generation for GMOS North longslit.  It will prefer a manual sequence if defined and otherwise generate an automatic sequence provided all required inputs are present and that we get an exposure time and count answer from ITC.  It looks at execution events to determine completed steps and take them into account when suggesting the next atom and possible future.

Screen Shot 2022-04-12 at 14 49 02

Caveats:

* Only GMOS North at the moment
* No smart GCAL so there's a stand-in flat step
* I don't know how to determine the acquisition exposure time so it is fixed
* There's no way to mark a dataset "bad" at the moment and consequently bad steps are not regenerated
* Doesn't yet drop in reacquisition sequences after time passes

### CODE REVIEW

**Comment by swalker2m** [comment]
I struggled mightily to make the code compact and understandable.  Unfortunately I feel that I wasn’t very successful in the end despite the effort.  I need to turn to OCS tasks for a while before coming back to add GMOS South.  Hopefully a bit of distance from the code will be beneficial.  If not, Rob will most certainly have better ideas when we implement this in the real ODB.

**Comment by rpiaggio** [comment]
Awesome! I'll start adding this to Explore. Is there a preloaded observation with all required parameters?

**Comment by swalker2m** [comment]
> Is there a preloaded observation with all required parameters?

Yes, `o-5` should work.

**Comment by swalker2m** [comment]
Another caveat, the spatial and wavelength offset, along with any other eventually configurable "Advanced" properties are all fixed at the moment.  This should be quick to address though.

File: modules/core/src/main/scala/lucuma/gen/Generator.scala [hunk 1]
```diff
@@ -3,50 +3,96 @@
...
-   * reached).
-   *
-   * @param observed a computation that decides when the observation is done
+   * Runs the Generator to make the execution config with its static config,
+   * acquisition sequence, and science sequence.
    */
-  def science[F[_]: Sync](
-    observed: F[Boolean]
-  ): Stream[F, AtomModel[StepModel[D]]]
+  def run[F[_]: Monad, S, D](
```
Comment (rpiaggio) [inline modules/core/src/main/scala/lucuma/gen/Generator.scala:55]: Can this be a method in `Generator` ?

Comment (swalker2m) [reply modules/core/src/main/scala/lucuma/gen/Generator.scala:55]: Yes, I wanted it to be there but was stymied.  How can I stipulate that `F[_]` have a `Monad` if I add move it to the instance?

**Review by rpiaggio — APPROVED** [review]
LGTM. Disclaimer: I didn't get into the deep GmosN generation logic.

File: modules/core/src/main/scala/lucuma/gen/Generator.scala (hunk 1 above)
Comment (rpiaggio) [reply modules/core/src/main/scala/lucuma/gen/Generator.scala:55]: We can have an `implicit F: Monad[F]` parameter in the method itself.

Comment (swalker2m) [reply modules/core/src/main/scala/lucuma/gen/Generator.scala:67]: Gah, of course!  Thanks, done.

**Pull request merged**